Makefile

```diff
@@ -30,8 +30,11 @@ lint:
 test:
 	pytest -v $(TEST)
 
+test-basic:
```

Comment:
how about we re-structure the way we call our tests a bit:

```Makefile
test: test-crud test-checkmode test-other

test-crud:
	pytest -v 'tests/test_crud.py::test_crud'

test-checkmode:
	pytest -v 'tests/test_crud.py::test_check_mode'

test-other:
	pytest -v -k 'not test_crud.py'
```

see [pytest docs how to specify test selection](https://docs.pytest.org/en/latest/usage.html#specifying-tests-selecting-tests)

and then the user still can do `make test` while CI does `make test-crud test-other` in most cases and once more with `test-checkmode`?

doesn't have to be in this PR tho if you get your version to work.

Reply:
oh, and this is missing the `$(TEST)` var we have above, so would need some more thought ;)

(or maybe just leave the old `make test` definition`)

Reply:
Can we drop the `$(TEST)` variable? Does anyone use it?
Also how about moving the static analysis tests into `lint` or `sanity`?
For now, i think, i'll "just" try add another step to do the check_mode tests isolated.

Reply:
Actually i'll leave the old `make test`, because that one runs all tests in one `pytest` instance, giving a better overview at the end.